src/foam/u2/view/ChoiceView.js

```diff
@@ -201,18 +203,29 @@ foam.CLASS({
 
       if ( this.dao ) this.onDAOUpdate();
 
-      this.start(this.selectSpec, {
-        data$: this.index$,
-        label$: this.label$,
-        alwaysFloatLabel: this.alwaysFloatLabel,
-        choices$: this.choices$,
-        placeholder$: this.placeholder$,
-        mode$: this.mode$,
-        size$: this.size$
-      })
-        .attrs({ name: this.name })
-        .enableClass('selection-made', this.index$.map((index) => index !== -1))
-      .end();
+      this.add(this.slot(function(mode){
+        if ( mode !== foam.u2.DisplayMode.RO ) {
+          return self.E()
+            .start(self.selectSpec, {
+              data$: self.index$,
+              label$: self.label$,
+              alwaysFloatLabel: self.alwaysFloatLabel,
+              choices$: self.choices$,
+              placeholder$: self.placeholder$,
+              mode$: self.mode$,
+              size$: self.size$
+            })
+              .attrs({ name: self.name })
+              .enableClass('selection-made', self.index$.map((index) => index !== -1))
+            .end();
+      
+        } else {
+          return self.E()
+            .start()
```

Comment:
Shouldn't need this extra start/end. `self.E().add(self.text$)` should do the trick.

Reply:
Done.